server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationMetaData.java

```diff
@@ -0,0 +1,291 @@
+    private static final ParseField LAST_COMMITTED_CONFIGURATION_FIELD = new ParseField("last_committed_config");
+    private static final ParseField LAST_ACCEPTED_CONFIGURATION_FIELD = new ParseField("last_accepted_config");
+
+    private static long term(Object[] termAndConfigs) {
+        return (long)termAndConfigs[0];
+    }
+
+    @SuppressWarnings("unchecked")
+    private static VotingConfiguration lastCommittedConfig(Object[] termAndConfig) {
+        List<String> nodeIds = (List<String>) termAndConfig[1];
+        return new VotingConfiguration(new HashSet<>(nodeIds));
+    }
+
+    @SuppressWarnings("unchecked")
+    private static VotingConfiguration lastAcceptedConfig(Object[] termAndConfig) {
+        List<String> nodeIds = (List<String>) termAndConfig[2];
+        return new VotingConfiguration(new HashSet<>(nodeIds));
+    }
+
+    private static final ConstructingObjectParser<CoordinationMetaData, Void> PARSER = new ConstructingObjectParser<>(
+            "coordination",
+            termAndConfigs -> new CoordinationMetaData(term(termAndConfigs), lastCommittedConfig(termAndConfigs),
+                    lastAcceptedConfig(termAndConfigs), Collections.emptySet()));
+    static {
+        PARSER.declareLong(ConstructingObjectParser.constructorArg(), TERM_PARSE_FIELD);
+        PARSER.declareStringArray(ConstructingObjectParser.constructorArg(), LAST_COMMITTED_CONFIGURATION_FIELD);
+        PARSER.declareStringArray(ConstructingObjectParser.constructorArg(), LAST_ACCEPTED_CONFIGURATION_FIELD);
+    }
+
+    public CoordinationMetaData(long term, VotingConfiguration lastCommittedConfiguration, VotingConfiguration lastAcceptedConfiguration,
+        Set<DiscoveryNode> votingTombstones) {
+        this.term = term;
+        this.lastCommittedConfiguration = lastCommittedConfiguration;
+        this.lastAcceptedConfiguration = lastAcceptedConfiguration;
+        this.votingTombstones = Collections.unmodifiableSet(new HashSet<>(votingTombstones));
+    }
+
+    public CoordinationMetaData(StreamInput in) throws IOException {
+        term = in.readLong();
+        lastCommittedConfiguration = new VotingConfiguration(in);
+        lastAcceptedConfiguration = new VotingConfiguration(in);
+        votingTombstones = Collections.unmodifiableSet(in.readSet(DiscoveryNode::new));
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static Builder builder(CoordinationMetaData coordinationMetaData) {
+        return new Builder(coordinationMetaData);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeLong(term);
+        lastCommittedConfiguration.writeTo(out);
+        lastAcceptedConfiguration.writeTo(out);
+        out.writeCollection(votingTombstones, (o, v) -> v.writeTo(o));
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        return builder
```

Comment:
can you add an example to the PR description how the new toXContent will look like?